plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/rating-filter/edit.tsx

```diff
@@ -120,54 +121,47 @@ const RatingFilterEdit = ( props: BlockEditProps< Attributes > ) => {
 	 * and filtered counts to get a list of options to display.
 	 */
 	useEffect( () => {
-		/**
-		 * Checks if a status slug is in the query state.
-		 *
-		 * @param {string} queryStatus The status slug to check.
-		 */
-
 		if ( filteredCountsLoading || isPreview ) {
 			return;
 		}
 
-		const orderedRatings =
-			! filteredCountsLoading &&
-			objectHasProp( filteredCounts, 'rating_counts' ) &&
-			Array.isArray( filteredCounts.rating_counts )
-				? [ ...filteredCounts.rating_counts ].reverse()
-				: [];
-
-		if ( orderedRatings.length === 0 ) {
+		if ( collectionFilters?.rating_counts?.length === 0 ) {
 			setDisplayedOptions( previewOptions );
 			return;
 		}
 
-		const newOptions = orderedRatings
-			.filter(
-				( item ) => isObject( item ) && Object.keys( item ).length > 0
-			)
-			.map( ( item ) => {
-				return {
-					label: (
-						<Rating
-							key={ item?.rating }
-							rating={ item?.rating }
-							ratedProductsCount={
-								showCounts ? item?.count : null
-							}
-						/>
-					),
-					value: item?.rating?.toString(),
-				};
-			} );
-
-		setDisplayedOptions( newOptions );
+		const minimumRating =
+			typeof minRating === 'string' ? parseFloat( minRating ) : 0;
+
+		/*
+		 * Process the ratings counts:
+		 * - Sort the ratings in descending order
+		 *   Todo: consider to handle this in the API request
+		 * - Filter out ratings below the minimum rating
+		 * - Map the ratings to the format expected by the filter component
+		 */
+		const productsRating = collectionFilters.rating_counts
+			.sort( ( a, b ) => b.rating - a.rating )
+			.filter( ( { rating } ) => rating >= minimumRating )
+			.map( ( { rating, count } ) => ( {
+				label: (
+					<Rating
+						key={ rating }
+						rating={ rating }
+						ratedProductsCount={ showCounts ? count : null }
+					/>
```

Comment:
I'm not sure if we can, Tung.
The techniques are different, and I wonder if it's doable to use`<RatingStars />` instead of `<Rating />`.

Points to consider:

* It will be rendered in the editor canvas as well as in the front-end
* `<Rating />` use the `WooCommerce` font to represent the stars
* `<RatingStars />` use SVG.

So if we are okay to use SVG in the canvas (I think it's OK) and also in the front-end, then I do believe we should use the `<RatingStars />`

Anyway, I'd like to do that in a follow-up.

Reply:
I don't see issues with other points.

> It will be rendered in the editor canvas as well as in the front-end

`<Rating>` isn't used on the frontend. We use the same markup but not the React component.

Reply:
> Anyway, I'd like to do that in a follow-up.

👍🏽

Reply:
> Let's remember that we'll need to render the empty star, too. Not a big deal, tho.

I don't think have empty stars in the new design. At least, I can't find them.

Reply:
you're right.

<img width="175" alt="image" src="https://github.com/user-attachments/assets/d5f20ba2-55ac-48d5-82c5-033d8a9fa114">

Reply:
We do some how load the style of `<Rating>` component on the frontend. I haven't figured out how yet.

Reply:
Not yet. I noted this in the PR description in the Disclaimer section.
The initial plan was to use this RatingStars component only in the block sidebar, a context different from the canvas. No a bid deal afaiks.

Reply:
> We do some how load the style of <Rating> component on the frontend. I haven't figured out how yet.

Found it, we're using the `<Rating>` component in preview and edit. MiniCssExtractPlugin will extract everything and put them into the main block CSS file.